cmake/openthread_extensions.cmake

```diff
@@ -0,0 +1,121 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+set(NRFXLIB_DIR $ENV{ZEPHYR_BASE}/../nrfxlib)
+assert_exists(NRFXLIB_DIR)
+include(${NRFXLIB_DIR}/common.cmake)
+
+set(OT_WORK_DIR ${CMAKE_CURRENT_SOURCE_DIR})
+
+# Store the configuration of the compiled OpenThread libraries
+# and set source and destination paths.
+function(openthread_libs_configuration_write)
+  unset(OPENTHREAD_SRC_DIR CACHE)
+  unset(OPENTHREAD_CONFIG_FILE CACHE)
+  unset(OPENTHREAD_LIB_FTD CACHE)
+  unset(OPENTHREAD_LIB_MTD CACHE)
+  unset(OPENTHREAD_LIB_RADIO CACHE)
+  unset(OPENTHREAD_LIB_CLI_FTD CACHE)
+  unset(OPENTHREAD_LIB_CLI_MTD CACHE)
+  unset(OPENTHREAD_LIB_NCP_FTD CACHE)
+  unset(OPENTHREAD_LIB_NCP_MTD CACHE)
+  unset(OPENTHREAD_LIB_RCP CACHE)
+  unset(OPENTHREAD_DST_DIR CACHE)
+
+  message(STATUS "OPENTHREAD_SRC_DIR=${OPENTHREAD_SRC_DIR}")
+  set(OPENTHREAD_SRC_DIR "${CMAKE_BINARY_DIR}/modules/openthread/build/src"
+    CACHE STRING "Directory containg the OpenThread source code.")
+  message(STATUS "CMAKE_BINARY_DIR=${CMAKE_BINARY_DIR}")
+  message(STATUS "OPENTHREAD_SRC_DIR=${OPENTHREAD_SRC_DIR}")
+
+  find_package(Git QUIET)
+  if(GIT_FOUND)
+    execute_process(
+      COMMAND           ${GIT_EXECUTABLE} rev-parse HEAD
+      WORKING_DIRECTORY ${OPENTHREAD_SRC_DIR}
```

Comment:
This will only work if the openthread module CMake code has been executed BEFORE this code.
Otherwise, `OPENTHREAD_SRC_DIR` may not be present.

In my test, I get:
```
cat build/openthread_lib_configuration.txt 
OpenThread_commit=
```

Also, you are getting the commits SHA of the build folder, I don't believe that is your intention.
The build folder can by anywere, even in `/tmp`